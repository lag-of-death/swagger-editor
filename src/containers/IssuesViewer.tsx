import * as React from "react";
import { connect } from "react-redux";
import styled from "styled-components";
import { border, borderRadius, boxShadow, padding } from "./shared";

const Issues = styled.div<{ hasIssues: boolean }>`
  display: flex;
  justify-content: ${({hasIssues}) => hasIssues ? "initial" : "center"};
  align-items: ${({hasIssues}) => hasIssues ? "initial" : "center"};
  opacity: ${({hasIssues}) => hasIssues ? 1 : 0.1};
  flex-grow: 0;
  flex-shrink: 1;
  ${padding}
  flex-basis: 100px;
  overflow: auto;
  margin: 10px;
  ${border}
  ${borderRadius}
  ${boxShadow}
  word-break: break-word;
`;

const toDivs = (issues: Array<{ message: string, path: string[] }>) => {
  return issues.map((msg) => {
    const path = msg.path.join(".");

    return (
      <div key={path}>
        <b>{path}</b>: {msg.message}
      </div>
    );
  });
};

export const IssuesViewer = ({issues}: { issues: Array<{ message: string, path: string[] }> }) => {
  const hasIssues = issues && !!issues.length;

  return (
    <Issues hasIssues={hasIssues}>{
      hasIssues
        ? <div>{toDivs(issues)}</div>
        : <div>CLICK ON A DOT TO LEARN ABOUT ISSUES...</div>
    }</Issues>
  );
};

export default connect((state: { issues: Array<{ message: string, path: string[] }> }) => {
  return {
    issues: state.issues,
  };
})(IssuesViewer);
