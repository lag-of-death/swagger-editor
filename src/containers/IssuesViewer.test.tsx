import { render } from "enzyme";
import "jest-styled-components";
import * as React from "react";
import { IssuesViewer } from "./IssuesViewer";

describe("IssuesViewer", () => {
  test("rendering when issues are supplied", () => {
    const component = render(
      <IssuesViewer issues={[{message: "warn", path: ["a", "b"]}]}/>,
    );

    expect(component).toMatchSnapshot();
  });

  test("rendering when issues are not supplied", () => {
    const component = render(
      <IssuesViewer issues={[]}/>,
    );

    expect(component).toMatchSnapshot();
  });
});
