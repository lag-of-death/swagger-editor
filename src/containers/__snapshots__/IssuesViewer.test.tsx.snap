// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IssuesViewer rendering when issues are not supplied 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  opacity: 0.1;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  padding: 6px;
  -webkit-flex-basis: 100px;
  -ms-flex-preferred-size: 100px;
  flex-basis: 100px;
  overflow: auto;
  margin: 10px;
  border: 6px solid black;
  border-radius: 6px;
  box-shadow: 2px 2px 2px black;
  word-break: break-word;
}

<div
  class="c0"
>
  <div>
    CLICK ON A DOT TO LEARN ABOUT ISSUES...
  </div>
</div>
`;

exports[`IssuesViewer rendering when issues are supplied 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: initial;
  -webkit-justify-content: initial;
  -ms-flex-pack: initial;
  justify-content: initial;
  -webkit-align-items: initial;
  -webkit-box-align: initial;
  -ms-flex-align: initial;
  align-items: initial;
  opacity: 1;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  padding: 6px;
  -webkit-flex-basis: 100px;
  -ms-flex-preferred-size: 100px;
  flex-basis: 100px;
  overflow: auto;
  margin: 10px;
  border: 6px solid black;
  border-radius: 6px;
  box-shadow: 2px 2px 2px black;
  word-break: break-word;
}

<div
  class="c0"
>
  <div>
    <div>
      <b>
        a.b
      </b>
      : warn
    </div>
  </div>
</div>
`;
